.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,48 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: Maven Publish
+
+on:
+  push: 
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: github
+        settings-path: ${{ github.workspace }}
+
+    - name: Format linter
+      working-directory: src/tigerbeetle-java
+      run: mvn -B formatter:validate --file pom.xml
+
+    - name: Install Zig
+      run: |
+        git submodule init
+        git submodule update
+        ./scripts/install.sh
+
+    - name: Build
+      working-directory: src/tigerbeetle-java
+      run: mvn -B package --file pom.xml
```

Comment:
Do tests get run somewhere in here as well?

Reply:
Yes, during the "package" phase, all validations and tests must pass.